doc/terminology.md

```diff
@@ -0,0 +1,86 @@
+
+A client is a program that intends to consume the VSS data points. This includes reading of sensors or actuating actors. A client intends to use capabilites offered by a vehicle.
+
+So typically this might be "apps" running on vehicle computers, the infotainment, or maybe even a user's personal device. 
+
+### Providers
+A provider is a program that makes sure interactions with the VSS data points are linked to the state of a vehicle.
+
+A provider can can supply either be a
+ *  *sensor-provider*, providing information for a sensor defined in VSS, i.e. a component that updates `Vehicle.Speed` based on the actual speed of an vehicle.
+ * An *actuator-provider*, trying to act on the target value of an actuator defined in VSS, i.e. a component that based on the target value of `Vehicle.Body.Trunk.Rear.IsOpen` tries to ensure that the actual state of the trunk (open or closed) matches the desired state set by clients.
+
+ ## Vehicle Signal Specification (VSS)
+ KUKSA.val handles datapoints that are defined using the COVESA Vehicle Specification (VSS).
+
+ VSS introduces a domain taxonomy for vehicle signals. It defines
+ * A syntax for defining vehicle signals in a structured manner.
+ * A catalog of standard signals (VSs standard catalogue) related to vehicles.
+
+It can be used as standard in automotive applications to communicate information around the vehicle, which is semantically well defined. 
+
+See the [the official VSS documentation](https://covesa.github.io/vehicle_signal_specification/) for all details. Here we will summarizes the terms and concepts most relevant for KUKSA.val.
+
+### Signal
+Everything we handle in KUKSA.val is defined as a using VSS. A signal in VSS is an entity identified by a dot-separated string in VSS, i.e. `Vehicle.Speed`.
+
+A signal can be of different types. VSS defines the following signal types:
+ - *Sensor*: Sensors are signals to read values of properties in a vehicle. Values of sensors typically change over time. Reading a sensor shall return the current actual value of the related property, e.g. the current speed of a vehicle. Example: `Vehicle.Speed`. [[Source]](https://covesa.github.io/vehicle_signal_specification/rule_set/data_entry/sensor_actuator/)
+ - *Actuator*: Actuators are signals that are  used to control the desired value of a property. _Every Actuator in VSS is also a Sensor_. Some properties in a vehicle may change instantly. A typical example is position of a seat. Reading a value of an actuator shall return the current actual value (i.e. the sensor trait of that actuator signal), e.g. the current position of the seat, rather than the wanted/desired position. Example: `set Vehicle.Cabin.Seat.Row1.Pos1.Position`. [[Source]](https://covesa.github.io/vehicle_signal_specification/rule_set/data_entry/sensor_actuator/)
```

Comment:
> _Every Actuator in VSS is also a Sensor_

I think this wording is very problematic.

It confuses `Sensor` with the _value_ of signal. A signal is either an `Actuator`, `Attribute` or `Sensor`. It cannot be two at the same time.

An `Actuator` signal has a `value` just _like_ a sensor (and an `Attribute` for that matter). And an `Actuator` behaves just _like_ a sensor with the exception that it's also possible to set it. But it is not a `Sensor`.

I think the same confusing wording is present above with **sensor-provider** (but it could be a typo, I'm not sure).

Reply:
The two types of providers are harder. I think that is also partially the point where we diverge in the authorisation PR. I don’t think “value” is defined/clear cut term in VSS. But I will try writing something about that in he other PR. 

For this PR let's see how to tackle it.

>> Every Actuator in VSS is also a Sensor
> I think this wording is very problematic.

I would stand by that statement, that an actuator can be seen as “sensor with benefits” in VSS,  I do agree that notion might not be obvious, shared by anyone. It is a poor choice here, if I expect a `sensor-provider` to also “feed” actuators and potentially attributes -> not obvious.

I was thinking about “signal-provider”, but then the question is, is that not the same as the “provide anything” in authorisation, i.e. it does not capture on an architecture level that I want to express, that this is a component that does not changes anything in the vehicle.

I think existing, “VSS terms” are not helping here, and introduced state-provider and actuation-provider as the two different classes of providers

Reply:
> I don’t think “value” is defined/clear cut term in VSS.

I agree with this when used as "Value provider". It is to generic, and sounds more like something out of the PR department.

> [...] it does not capture on an architecture level that I want to express, that this is a component that does not changes anything in the vehicle. 
> I think existing, “VSS terms” are not helping here, and introduced state-provider and actuation-provider as the two different classes of providers

I think "state provider" is basically as generic as "value provider" (and sounds like a state (government) provider).

How do you feel about "data provider"? That should make it clear that it's not changing anything in the vehicle, and it feels a lot more natural to speak of in this context. I think someone at FOSDEM even spontaneously referred to them as "data providers".

Reply:
I am quite happy with something like "data provider" and "actuation provider". I see a value in not using the term "sensor" as it is not always a "real sensor", it could for instance be a calculated/estimated value.

Reply:
Changed to data-provider now